server/asset/eth/tokener.go

```diff
@@ -44,11 +44,11 @@ func newTokener(ctx context.Context, assetID uint32, net dex.Network, be bind.Co
 		return nil, err
 	}
 
-	if token.ver != 0 {
+	if token.ver != 1 {
 		return nil, fmt.Errorf("only version 0 contracts supported")
 	}
 
-	es, err := erc20v0.NewERC20Swap(swapContract.Address, be)
+	es, err := erc20v1.NewERC20Swap(swapContract.Address, be)
```

Comment:
I had always intended for the server to only support a single version at a time, and I still do. But there's probably an argument for making it switchable via configuration. I don't know.

Reply:
Although it would seem crazy to us, a user will update in the middle of a swap. Or perhaps a redeem or refund that forgot about. It would be responsible to continue to support old versions and deprecate slowly.

Reply:
I would expect the client would be able to refund or auto-redeem those on their own.

Slightly related, which I am still intending to address myself: https://github.com/decred/dcrdex/issues/1583

Reply:
I want to add that we are continuously making changes with new wallets, and probably more, that add data to the db that makes downgrading impossible but does not error if you try. So, a user may just be really confused and out of luck if they accidentally upgraded with an old swap. They would need to restore from seed with an older version.

Reply:
Oh yeah, we are handling both versions for the client. Please ignore my comment.

Reply:
Oh, but yeah, I guess markets will just need to be suspended for makerlocktime to ensure no new initiations? And hopefully refunds will still be ok, I think ok. Should be ok.